addon/ng2/models/config.ts

```diff
@@ -166,6 +167,24 @@ export class CliConfig {
 
   public static fromProject(): any {
     const configPath = CliConfig._configFilePath();
-    return configPath ? require(configPath) : {};
+
+    if (!configPath) {
+      return {};
+    }
+
+    let config = require(configPath);
+
+    if (config.defaults.sourceDir || config.defaults.prefix) {
+      config.apps[0].root = config.apps[0].root || config.defaults.sourceDir;
+      config.apps[0].prefix = config.apps[0].prefix || config.defaults.prefix;
+      
+      let message = 'The "defaults.prefix" and "defaults.sourceDir" properties of angular-cli.json';
+      message += 'are deprecated in favor of "apps[0].root" and "apps[0].prefix".';
+      message += ' Please update in order to avoid errors in future versions of angular-cli.';
+
+      console.error(chalk.yellow(message);
```

Comment:
Please refactor like this:

```typescript
console.error(chalk.yellow(
    'The "defaults.prefix" and "defaults.sourceDir" properties of angular-cli.json'
  + 'are deprecated in favor of "apps[0].root" and "apps[0].prefix".'
  + ' Please update in order to avoid errors in future versions of angular-cli.'));
```